third_party/amd/lib/TritonAMDGPUTransforms/BlockPingpong.cpp

```diff
@@ -1302,11 +1343,19 @@ void Pingponger::getDotPingponged() {
       LDBG("Currently only support num_warp=8 for async PP");
       return;
     }
+    if (numStages == 3 && dotOps.size() == 1 && tileSize == mediumTile && aShape[1] == 32 && elemWidth == 16) {
```

Comment:
@guacamoleo it's guarded here, only enabled with supported tile size (and expected to be performant) and data type. Checking num_stages=3 is just first step to check it.